core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMap.java

```diff
@@ -31,7 +31,8 @@
   /**
    * It is called to load the data map to memory or to initialize it.
    */
-  void init(String filePath) throws MemoryException, IOException;
+  void init(String blockletIndexPath, String customIndexPath, String segmentId)
```

Comment:
The `filepath` supposed to be either index folder name or index file name, so I don't think this extra information is required here.
And also `blockletIndexPath` is not supposed passed as we have carbonIndex exists in other datamap and we supposed to use it.

Reply:
For Min Max Index creation like segment properties and other things i am taking input from regular carbonindex file too. So by design we can have one parameter as primitive index path other can be of the new custom index file path.

Reply:
it should be independent of other indexes

Reply:
In this example Along with Min and Max Information i am keeping few more information for building the BlockLet. Both indexes are independent but with the current example implementation i read the Min and Max index and and then read the carbonindex index also in order to get the column cardanality and segmentproperties. These values are used to form the blocklet used for pruning.